### PR TITLE
[iOS] Allow compat iOS buttons to resize image and to respect spacing and padding

### LINKED ISSUE #9734: Button with image not scaling to HeightRequest on iOS and Android
### Description

Button with an image fails to correctly scale the source image to the size of the button this results in the full size image being displayed in Android and a skewed image in iOS

iOS
Simulator Screen Shot - iPad Pro (12 9-inch) (5th generation) - 2022-08-29 at 17 43 58

Android
Screenshot_1661759181

WinUI
Screenshot 2022-08-29 180009

### Steps to Reproduce

Create a new example Maui App

Add the following line of xaml to MainPage.xaml

```
<Button
                ImageSource="dotnet_bot.png"
                Text="Im a happy button"
                HeightRequest="40"
                WidthRequest="200"/>
```

Run the solution, works correctly in WinUI and results in the full sized image next to the button in Android and the image skewed and distorted but with the correct height in iOS 

I expect this to correctly render the image in the button to the left of the text in the button using the heightrequest to scale down to the requested height maintaining the images aspect ratio, see WinUI example above where the image and button are rendered correctly

### Version with bug

6.0.486 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Android

### Affected platform versions

iOS 15.5, Android 33

### Did you find any workaround?

Reduce the image size manually to be the correct size desired on a device

### Relevant log output

_No response_

### LINKED ISSUE #18242: Button ImageSource not Scaling as expected.
### Description

Using a Button and adding a ImageSource, svg, the image is not scaled as expected. 
According to Documentation: While **Button** defines an **ImageSource** property, that allows you to display a image on the **Button**, this property is intended to be used when displaying a small icon next to the **Button** text.

On the screenshots below the first component is a Button with ImageSource, and the Second is a HorizontalStackLayout with ImageButton and Label, set up with same HeightRequest. 

iOS:
image

Android:
image

And since I am using it in a popup dialog from CommunityToolkit i'm unable to use method 2, since GestureRecognizers are not working. 

xaml to reproduce:
image

### Steps to Reproduce

1. Remove content of VerticalStackLayout in Maui sample app.
2. Add screenshot of xaml as content. (could not get 'code' to work)
            

### Link to public reproduction project repository

_No response_

### Version with bug

8.0.0-rc.2.9373

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Android

### Affected platform versions

_No response_

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by tj-devel709**
### Description of Change

This PR is a continuation of https://github.com/dotnet/maui/pull/20953. 

After a lot of discussion and testing, it seems the best solution to fix iOS buttons is to get the compatibility APIs in a better state (able to resize, respect the spacing, and respect the padding) and then introduce the iOS 15+ new Configuration APIs. A good amount of work was put in to place the title and image exactly where we want using precise values for the TitleEdgeInsets and ImageEdgeInsets but we noticed that there was some limitation at seemingly random times where the image and title would be several pixels from where we placed them. Since the iOS 15+ Configuration API takes care of placing the image and title in different image placements, that will have improved behavior.

Below is a demo of the button behavior on iOS. The left simulator is prior to this PR and the right displays the changes from this PR. Pay attention to the resizing of the images, the placement/centering, and then respecting the padding and spacing (distance between the image and text).

~~https://github.com/dotnet/maui/assets/50846373/b85a8bad-8e0c-456a-8632-b53795ec1a79~~

[image
](https://github.com/dotnet/maui/assets/50846373/00e75b42-e62f-4c62-bfff-2bdf58e3e0bd)

Code for the sample above: https://gist.github.com/tj-devel709/716c675a6324c2757f2c84354312a51c

### Issues Fixed

Fixes #9734
Fixed #18242

### CODE REVIEW

File: src/Controls/samples/Controls.Sample.UITests/Issues/Issue18242_2.xaml [hunk 1]
```diff
@@ -0,0 +1,70 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
+             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
+             x:Class="Maui.Controls.Sample.Issues.Issue18242_2">
+  <ScrollView>
```
Comment (jsuarezruiz) [inline src/Controls/samples/Controls.Sample.UITests/Issues/Issue18242_2.xaml:5]: fix-18232_2

**Comment by tj-devel709** [comment]
https://github.com/dotnet/maui/assets/50846373/00e75b42-e62f-4c62-bfff-2bdf58e3e0bd

After many discussions, here is what the buttons will do on iOS. If the image is too large for the button, it will be resized to the max size that can fit into the button without regard to the text or the spacing. If the button has text or spacing, the button content will be larger than what it can hold and will spill out into the padding or even outside the button. The most correct way to deal with this is to use the right sized image for your button in your project. 

The reason why we are going forward with this direction is that on iOS 15+, the Configuration APIs for UIButton do not contain a way to simply ignore/hide the title where on iOS 14 and below, we could technically move the title out of the button if it did not fit. This leaves us with different options, prioritize the title's area and give the image the rest of the space - but this leads us to scenarios where the image can be 1x1 pixel and cause other issues. We could also have some sort of cap and say the image can take up half of the space of the button, but this is not always desired as well and can lead to unexpected behavior. Other options make things less clear to the developer about what exactly is going on in the layout of the button. The current decision allows the iOS 14 and below and iOS 15 and above (using different supported APIs from Apple) to have more similar behavior and avoid further surprises for the developer.

**Comment by jsuarezruiz** [comment]
> newButtoniOSDemo.mov 
> After many discussions, here is what the buttons will do on iOS. If the image is too large for the button, it will be resized to the max size that can fit into the button without regard to the text or the spacing. If the button has text or spacing, the button content will be larger than what it can hold and will spill out into the padding or even outside the button. The most correct way to deal with this is to use the right sized image for your button in your project.
> 
> The reason why we are going forward with this direction is that on iOS 15+, the Configuration APIs for UIButton do not contain a way to simply ignore/hide the title where on iOS 14 and below, we could technically move the title out of the button if it did not fit. This leaves us with different options, prioritize the title's area and give the image the rest of the space - but this leads us to scenarios where the image can be 1x1 pixel and cause other issues. We could also have some sort of cap and say the image can take up half of the space of the button, but this is not always desired as well and can lead to unexpected behavior. Other options make things less clear to the developer about what exactly is going on in the layout of the button. The current decision allows the iOS 14 and below and iOS 15 and above (using different supported APIs from Apple) to have more similar behavior and avoid further surprises for the developer.

I think we should include some notes in the docs.

cc @davidbritch

**Review by jsuarezruiz — APPROVED** [review]

**Comment by PureWeen** [comment]
Rolled back here: https://github.com/dotnet/maui/pull/22300

**Pull request merged**